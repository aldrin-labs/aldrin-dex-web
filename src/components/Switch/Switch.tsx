import * as React from 'react'
import styled from 'styled-components'

interface Props {
  onClick?: Function
  values?: string[]
  isActive?: boolean
}

export default class Switch extends React.Component<Props> {
  render() {
    const { onClick, values, isActive } = this.props
    const [first, second] = values || ['', '']

    return (
      <Container>
        {first && <Desc>{first}</Desc>}
        <Label>
          <Input type="checkbox" onClick={onClick} checked={isActive} />
          <Slider />
        </Label>
        {second && <Desc>{second}</Desc>}
      </Container>
    )
  }
}

const Desc = styled.span`
  font-family: Roboto;
  font-size: 1em;
  margin: 0 10px;
  color: #fff;
  font-weight: 500;
`

const Container = styled.div`
  display: flex;
  align-items: center;
`

const Slider = styled.span`
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #7fcac380;
  transition: 0.4s;
  border-radius: 34px;

  & :before {
    position: absolute;
    content: '';
    height: 20px;
    width: 20px;
    left: -5px;
    bottom: -3px;
    background-color: #4ed8da;
    transition: 0.4s;
    border-radius: 50%;
  }
`

const Input = styled.input`
  :checked + ${Slider}:before {
    -webkit-transform: translateX(26px);
    -ms-transform: translateX(26px);
    transform: translateX(26px);
  }
`

// & :checked + ${Slider} {
//   background-color: #2196f3;
// }
// & :focus + ${Slider} {
//   box-shadow: 0 0 1px #2196f3;
// }

const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 34px;
  height: 14px;

  & ${Input} {
    display: none;
  }
`
