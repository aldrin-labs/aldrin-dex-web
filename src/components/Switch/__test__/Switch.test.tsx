import * as React from 'react'
import { shallow, mount } from 'enzyme'
import Switch from '../Switch'

describe('Switch', () => {
  describe('render', () => {
    it('basic', () => {
      const wrapper = shallow(<Switch />)

      expect(wrapper).toMatchSnapshot()
    })

    it('with values', () => {
      const wrapper = shallow(<Switch values={['ON', 'OFF']} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('if isActive', () => {
      const wrapper = shallow(<Switch values={['ON', 'OFF']} isActive />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  it('onClick Switch', () => {
    const onClick = jest.fn()
    const wrapper = mount(
      <Switch values={['ON', 'OFF']} onClick={onClick} isActive />
    )
    wrapper
      .find('input')
      .props()
      .onClick()

    expect(onClick).toBeCalled()
  })
})
