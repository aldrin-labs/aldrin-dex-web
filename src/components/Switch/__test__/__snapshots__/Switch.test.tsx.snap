// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch render basic 1`] = `
<file__Container>
  <file__Label>
    <file__Input
      type="checkbox"
    />
    <file__Slider />
  </file__Label>
</file__Container>
`;

exports[`Switch render if isActive 1`] = `
<file__Container>
  <file__Desc>
    ON
  </file__Desc>
  <file__Label>
    <file__Input
      checked={true}
      type="checkbox"
    />
    <file__Slider />
  </file__Label>
  <file__Desc>
    OFF
  </file__Desc>
</file__Container>
`;

exports[`Switch render with values 1`] = `
<file__Container>
  <file__Desc>
    ON
  </file__Desc>
  <file__Label>
    <file__Input
      type="checkbox"
    />
    <file__Slider />
  </file__Label>
  <file__Desc>
    OFF
  </file__Desc>
</file__Container>
`;
